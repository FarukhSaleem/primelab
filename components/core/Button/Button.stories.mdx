import Button from "./Button.tsx"

# Button
<Canvas>
    <Story
        name="Default"
        args={{}}>
    <Button label="ABC"/>
    </Story>
</Canvas>