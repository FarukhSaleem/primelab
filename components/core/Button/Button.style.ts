import {makeStyles} from "@mui/styles";

export const useStyles = makeStyles((theme: any) => ({
  button: {
    padding: '5px',
    marginTop: 3,
    marginBottom: 2, 
  },
}))
export default useStyles;