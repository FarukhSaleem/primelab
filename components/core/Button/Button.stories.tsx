import React from 'react';
import { Box } from '@mui/material';
import { ComponentMeta } from '@storybook/react'

import Button from './Button';

export default {
  title: 'Component/Button',
  component: Button,
  decorators: [
    Story => (
      <Box maxWidth='lg'>
        <Story />
      </Box>
    )
  ]
} as ComponentMeta<typeof Button>

export const Primary = ({label='Button Primary'}: {label: string}) => <Button label={label} />;