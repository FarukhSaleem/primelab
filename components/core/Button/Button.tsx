import React from "react";
import MaterialButton from "@mui/material/Button";
import { Props } from "./Button.type";
import useStyles from "./Button.style";

const Button = ({ label, ...restProps }: Props): JSX.Element => {
  const classes = useStyles();
  return (
    <MaterialButton className={classes.button} {...restProps} data-testid="submit-button-custom" >
      {label}
    </MaterialButton>
  );
};
export default Button;
