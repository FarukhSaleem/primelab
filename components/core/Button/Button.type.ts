import { ButtonProps } from "@mui/material/Button";

export type Props = {
  label: string;
} & ButtonProps;
