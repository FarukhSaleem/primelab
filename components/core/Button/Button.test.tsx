import React from "react";
import { render } from "@testing-library/react";
import Button from "./Button";
import { Props } from "./Button.type";

function renderButton(props: Props) {
  return render(<Button {...props} />);
}

describe("<Button />", () => {
  test("It should render submit button", async () => {
    const { getByTestId } = renderButton({ label: "Submit Button", className: "MyTestButton" });
    const buttonElement = getByTestId("submit-button-custom");
    expect(buttonElement).toMatchSnapshot();
    expect(buttonElement.innerHTML).toContain('Submit Button')
  });
});
