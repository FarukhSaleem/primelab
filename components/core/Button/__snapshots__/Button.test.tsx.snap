// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> It should render submit button 1`] = `
<button
  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MyTestButton css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  data-testid="submit-button-custom"
  tabindex="0"
  type="button"
>
  Submit Button
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
